pkg/terraform/reference.go

```diff
@@ -133,33 +132,43 @@ func (r Reference) RefersTo(a Reference) bool {
 }
 
 func (r *Reference) SetKey(key cty.Value) {
+	if key.IsNull() || !key.IsKnown() {
```

Comment:
Wouldn't this panic if key was actually null? In that case the `!key.IsKnown()` would panic.

Reply:
ping @nikpivkin

Reply:
@simar7 The right operand of a boolean expression is evaluated [conditionally](https://go.dev/ref/spec#Logical_operators). If the `key` is null, then the left operand of the expression in the condition will be true and the function will exit, without checking whether the `key` is known.  And in any case, why should this check cause panic?

Reply:
I see. I didn’t know the conditional evaluation of Go for such a case. This seems to follow the short circuit behaviour. 

Thanks this works fine then.